src/lsdb/dask/join_catalog_data.py

```diff
@@ -210,16 +243,24 @@ def join_catalog_data_through(
             association.hc_structure.catalog_info.join_column,
             suffixes,
         )
-    alignment = align_trees(
-        left.hc_structure.pixel_tree, right.hc_structure.pixel_tree, alignment_type=PixelAlignmentType.INNER
-    )
+
+    if right.margin is None:
+        warnings.warn("Right catalog does not have a margin cache. Results may be inaccurate", RuntimeWarning)
```

Comment:
These warnings are being thrown a lot in the unit tests. Are you planning to add margins to additional data sets (maybe as a follow-up), or expect the warnings in the unit tests?